app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt

```diff
@@ -173,4 +186,51 @@ class NewOnboardingFragment :
     override fun navigateToContinueCreateWallet() {
         replaceFragment(ContinueOnboardingFragment.create())
     }
+
+    private fun setLoadingAnimationState(isScreenLoading: Boolean) {
+        with(binding) {
+            TransitionManager.beginDelayedTransition(root)
+            loadingAnimationView.isVisible = isScreenLoading
+            animationView.apply {
+                if (isScreenLoading) {
+                    startCreationProgressJob()
+                    playAnimation()
+                } else {
+                    creationProgressJob?.cancel()
+                    cancelAnimation()
+                }
+            }
+        }
+    }
+
+    private fun startCreationProgressJob() {
+        creationProgressJob = listOf(
+            TimerState(R.string.onboarding_loading_title),
+            TimerState(R.string.onboarding_loading_title_1, 33),
+            TimerState(R.string.onboarding_loading_title_2, 55),
+            TimerState(R.string.onboarding_loading_title_3, 100),
+        ).asSequence()
+            .asFlow()
+            .onEach {
+                with(binding) {
+                    textViewCreationTitle.setText(it.titleRes)
+                    val hasProgress = it.progress > 0
+                    textViewCreationMessage.isVisible = !hasProgress
+                    progressBarCreation.isVisible = hasProgress
+                    progressBarCreation.setProgress(it.progress, true)
+                }
+                delay(2.seconds.inWholeMilliseconds)
+            }
+            .launchIn(viewLifecycleOwner.lifecycleScope)
```

Comment:
you could make the progressbar as `indeterminate` and that's it. 
In this case the actual data maybe still loading, you just hardcoded the animation.
I think it's wrong